src/librustc_lint/unused.rs

```diff
@@ -35,7 +35,9 @@ declare_lint! {
 }
 
 #[derive(Copy, Clone)]
-pub struct UnusedMut;
+pub struct UnusedMut {
+    borrowck_results: Vec<Rc<BorrowCheckResult>>
```

Comment:
It seems that there are no uses of this field -- looking at the rest of this file, it seems like it still reads from `tcx.used_mut_nodes`. Once you add the `pub` keyword to the above file, of course, we'll get to this crate, at which point it won't compile, since `tcx.used_mut_nodes` has been removed.